components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -678,8 +678,15 @@ public OAuthRevocationResponseDTO revokeTokenByOAuthClient(OAuthRevocationReques
 
                 } else if (accessTokenDO != null) {
                     if (revokeRequestDTO.getConsumerKey().equals(accessTokenDO.getConsumerKey())) {
+                        // Resolve the application tenant domain.
+                        String tenantDomain;
+                        if (StringUtils.isNotEmpty(PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                .getApplicationResidentOrganizationId())) {
+                            tenantDomain = IdentityTenantUtil.getTenantDomain(accessTokenDO.getTenantID());
+                        } else {
+                            tenantDomain = IdentityTenantUtil.getTenantDomain(IdentityTenantUtil.getLoginTenantId());
+                        }
```

Comment:
```suggestion
                            tenantDomain = IdentityTenantUtil.getTenantDomain(IdentityTenantUtil.getLoginTenantId());
                        if (StringUtils.isNotEmpty(PrivilegedCarbonContext.getThreadLocalCarbonContext()
                                .getApplicationResidentOrganizationId())) {
                            tenantDomain = IdentityTenantUtil.getTenantDomain(accessTokenDO.getTenantID());
                        }
```


WDYT?

Reply:
Swapped the if else blocks as per the offline chat